services/dns/client/client.go

```diff
@@ -0,0 +1,104 @@
+/* Copyright (c) 2022 Snowflake Inc. All rights reserved.
+
+   Licensed under the Apache License, Version 2.0 (the
+   "License"); you may not use this file except in compliance
+   with the License.  You may obtain a copy of the License at
+
+     http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing,
+   software distributed under the License is distributed on an
+   "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+   KIND, either express or implied.  See the License for the
+   specific language governing permissions and limitations
+   under the License.
+*/
+
+// Package client provides the client interface for 'dns'
+package client
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"os"
+
+	"github.com/google/subcommands"
+
+	"github.com/Snowflake-Labs/sansshell/client"
+	pb "github.com/Snowflake-Labs/sansshell/services/dns"
+	"github.com/Snowflake-Labs/sansshell/services/util"
+)
+
+const subPackage = "dns"
+
+func init() {
+	subcommands.Register(&dnsCmd{}, subPackage)
+}
+
+func setup(f *flag.FlagSet) *subcommands.Commander {
+	c := client.SetupSubpackage(subPackage, f)
+	c.Register(&lookupCmd{}, "")
+	return c
+}
+
+type dnsCmd struct{}
+
+func (*dnsCmd) Name() string { return subPackage }
+func (p *dnsCmd) Synopsis() string {
+	return client.GenerateSynopsis(setup(flag.NewFlagSet("", flag.ContinueOnError)), 2)
+}
+func (p *dnsCmd) Usage() string {
+	return client.GenerateUsage(subPackage, p.Synopsis())
+}
+func (*dnsCmd) SetFlags(f *flag.FlagSet) {}
+
+func (p *dnsCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	c := setup(f)
+	return c.Execute(ctx, args...)
+}
+
+type lookupCmd struct{}
+
+func (*lookupCmd) Name() string     { return "lookup" }
+func (*lookupCmd) Synopsis() string { return "Executes a DNS lookup" }
+func (*lookupCmd) Usage() string {
+	return `lookup:
+    Performs a simple DNS lookup for the provided hostname on the remote server.
+`
+}
+
+func (p *lookupCmd) SetFlags(f *flag.FlagSet) {}
+
+func (p *lookupCmd) Execute(ctx context.Context, f *flag.FlagSet, args ...interface{}) subcommands.ExitStatus {
+	state := args[0].(*util.ExecuteState)
+	if f.NArg() == 0 {
```

Comment:
if f.Narg() != 1

i.e. this catches people adding extra things that don't parse then too

Reply:
done in 3f31c60002e0aba6c8a2ae49ccabc7211d1db494